code/modules/mob/living/simple_animal/roach.dm

```diff
@@ -0,0 +1,247 @@
+		//If there is food, climb on it (using pixel_x and pixel_y manipulation)
+		animate(src, pixel_x = F.pixel_x + rand(-4,4), pixel_y = F.pixel_y + rand(-4,4), rand(10,20), 1)
+
+		layer = F.layer + 0.01
+
+		if(flying)
+			stop_flying(anim = 0)
+
+		spawn()
+			turns_since_move -= rand(5,20) //Stay here for a while. turns_since_move is set to 0 immediately after this proc, so the spawn() is required.
+
+			if((last_laid_eggs + 30 SECONDS < world.time)) //Base cooldown is 30 seconds. The actual cooldown is actually slightly more than that
+				sleep(rand(1,5) SECONDS)
+
+				//And yeah, roaches can lay eggs on their own eggs. This is kinda intended
+				lay_eggs()
+
+				if(F && F.reagents)
+					F.reagents.add_reagent("toxin", rand(0.2,0.6)) //Add some toxin to the food
+
+		return //Don't do anything after that
+
+	//Then, check for any trash
+	for(var/obj/item/trash/T in loc)
+		//If there is trash, climb under it (using pixel_x, pixel_y and layer manipulation)
+		animate(src, pixel_x = T.pixel_x, pixel_y = T.pixel_y, rand(10,30), 1)
+
+		layer = T.layer - 0.01
+
+		if(flying)
+			stop_flying(anim = 0)
+
+		spawn()
+			turns_since_move -= rand(5,20) //Stay here for a while
+
+			if((last_laid_eggs + 30 SECONDS < world.time) && prob(25)) //25% chance of laying eggs
+				sleep(rand(1,5) SECONDS)
+
+				lay_eggs()
+
+		return
+
+	//If there's no food, check for any walls to climb on
+	var/turf/simulated/wall/T = dest //If we attempted to move into a wall
+	if(istype(T))
+		var/check_dir = get_dir(src, dest)
+
+		//Climb on it!
+		var/new_px = rand(-8,8) + cos(dir2angle(check_dir)) * 32
+		var/new_py = rand(-8,8) + sin(dir2angle(check_dir)) * 32
+
+		//Modify pixel_x and pixel_y to make it look like the cockroach is on the wall
+		animate(src, pixel_x = new_px, rand(5,15), 1, ELASTIC_EASING)
+		animate(src, pixel_y = new_py, rand(5,15), 1, ELASTIC_EASING)
+
+		return
+
+	//No food, trash, walls or anything - just modify our pixel_x and pixel_y
+	animate(src, pixel_x = rand(-20,20), pixel_y = rand(-20,20), (flying ? 5 : 15) , 1) //This animation takes 1.5 seconds, or 0.5 if flying
+
+/mob/living/simple_animal/cockroach/Move()
+	..()
+
+	if(!flying)
+		layer = initial(layer) //Since cucarachas can hide under trash (which modifies their layer), this is kinda necessary
+
+/mob/living/simple_animal/cockroach/adjustBruteLoss() //When receiving damage
+	..()
+
+	if(health > 0) //Still alive
+		start_flying()
+
+		if(usr)
+			var/mob/user = usr
+			var/turf/new_location = get_step(src, turn(get_dir(src,user),180)) //Walk away from the source of the damage
+
+			wander_move(new_location)
+
+		spawn(rand(4,10) SECONDS)
+			stop_flying()
+
+/mob/living/simple_animal/cockroach/proc/start_flying(var/anim = 1)
+	if(isUnconscious()) return
+
+	speed = -4
+	turns_since_move = 5 //Remove any delay
+
+	icon_state = "cockroach_fly"
+
+	flying = 1
+	speak_chance = 5
+
+	turns_per_move = 1
+
+	response_help  = "attepts to pet"
```

Comment:
"attepts" should be "attempts"
am i doing this right? am i allowed to bitch about typos?